syntax = "proto3";

package logika;

enum ArchiveTypeEnum
{
    ARCHIVE_TYPE_UNDEFINED  = 0;
    ARCHIVE_TYPE_CONTROL    = 1;
    ARCHIVE_TYPE_DAY        = 2;
    ARCHIVE_TYPE_DECADE     = 3;
    ARCHIVE_TYPE_DIAGS_LOG  = 4;
    ARCHIVE_TYPE_ERRORS_LOG = 5;
    ARCHIVE_TYPE_HOUR       = 6;
    ARCHIVE_TYPE_MONTH      = 7;
    ARCHIVE_TYPE_PARAMS_LOG = 8;
    ARCHIVE_TYPE_POWER_LOG  = 9;
    ARCHIVE_TYPE_TURN       = 10;
    ARCHIVE_TYPE_MINUTE     = 11;
    ARCHIVE_TYPE_HALF_HOUR  = 12;
}

enum BusTypeEnum
{
    BUS_TYPE_UNDEFINED  = 0;
    BUS_TYPE_RS_BUS     = 1;
    BUS_TYPE_SP_BUS     = 2;
}

enum TagKindEnum
{
    TAG_KIND_UNDEFINED   = 0;
    TAG_KIND_PARAMETER   = 1;
    TAG_KIND_INFO        = 2;
    TAG_KIND_REALTIME    = 3;
    TAG_KIND_TOTAL_CTR   = 4;
}

enum DataTypeEnum
{
    DATA_TYPE_UNDEFINED     = 0;
    DATA_TYPE_BYTE          = 1;
    DATA_TYPE_INT32         = 2;
    DATA_TYPE_INT64         = 3;
    DATA_TYPE_SINGLE        = 4;
    DATA_TYPE_DOUBLE        = 5;
    DATA_TYPE_STRING        = 6;
    DATA_TYPE_BYTE_ARRAY    = 7;
    DATA_TYPE_INT32_ARRAY   = 8;
    DATA_TYPE_STRING_ARRAY  = 9;
    DATA_TYPE_OBJECT_ARRAY  = 10;
}

enum VarTypeEnum
{
    VAR_TYPE_UNDEFINED  = 0;
    VAR_TYPE_SP         = 1;
    VAR_TYPE_NS         = 2;
    VAR_TYPE_AVG        = 3;
    VAR_TYPE_G          = 4;
    VAR_TYPE_M          = 5;
    VAR_TYPE_P          = 6;
    VAR_TYPE_DP         = 7;
    VAR_TYPE_T          = 8;
    VAR_TYPE_TI         = 9;
    VAR_TYPE_V          = 10;
    VAR_TYPE_W          = 11;
    VAR_TYPE_AUX_INT    = 12;
}

enum InternalTypeEnum
{
    INTERNAL_TYPE_UNDEFINED         = 0;
    INTERNAL_TYPE_R32               = 1;
    INTERNAL_TYPE_TRIPLE_R32        = 2;
    INTERNAL_TYPE_TIME              = 3;
    INTERNAL_TYPE_DATE              = 4;
    INTERNAL_TYPE_MM_DD_DATE        = 5;
    INTERNAL_TYPE_BIT_ARRAY32       = 6;
    INTERNAL_TYPE_BIT_ARRAY24       = 7;
    INTERNAL_TYPE_BIT_ARRAY16       = 8;
    INTERNAL_TYPE_BIT_ARRAY8        = 9;
    INTERNAL_TYPE_DB_ENTRY          = 10;
    INTERNAL_TYPE_DB_ENTRY_BYTE     = 11;
    INTERNAL_TYPE_U8                = 12;
    INTERNAL_TYPE_I32R32            = 13;
    INTERNAL_TYPE_MM_HH_TIME        = 14;
    INTERNAL_TYPE_NS_RECORD         = 15;
    INTERNAL_TYPE_IZM_RECORD        = 16;
    INTERNAL_TYPE_ARCHIVE_STRUCT    = 17;
    INTERNAL_TYPE_MODEL_CHAR        = 18;
    INTERNAL_TYPE_U24               = 19;
    INTERNAL_TYPE_SERVICE_RECORD_TS = 20;
}

enum X6TagTypeEnum
{
    X6_TAG_TYPE_UNDEFINED   = 0;
    X6_TAG_TYPE_ARRAY       = 1;
    X6_TAG_TYPE_STRUCTURE   = 2;
    X6_TAG_TYPE_TAG         = 3;
}
